### PR TITLE
Added possibility to specify TLS cert for smtpd

### PR DESCRIPTION
**Pull request by albgus**
Added parameters to configure smtpd_tls_cert_file and smtpd_tls_key_file
Also added checks for TLS related parameters with default configuration

### CODE REVIEW

**Comment by Phil-Friderici** [comment]
@gusson thanks for your work :+1:

What about adding an option to roll out the TLS certs as well ? Think that would make this complete ;)

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -217,6 +217,14 @@
     it { should contain_file('postfix_main.cf').without_content(/^mailbox_command =/) }
     it { should contain_file('postfix_main.cf').without_content(/^relay_domains =/) }
     it { should contain_file('postfix_main.cf').without_content(/^transport_maps/) }
+    it { should contain_file('postfix_main.cf').without_content(/^smtp_tls_mandatory_protocols/) }
+    it { should contain_file('postfix_main.cf').without_content(/^smtp_tls_protocols/) }
+    it { should contain_file('postfix_main.cf').without_content(/^smtp_tls_security_level/) }
+    it { should contain_file('postfix_main.cf').without_content(/^smtpd_tls_mandatory_protocols/) }
+    it { should contain_file('postfix_main.cf').without_content(/^smtpd_tls_protocols/) }
+    it { should contain_file('postfix_main.cf').without_content(/^smtpd_tls_security_level/) }
```
Comment (Phil-Friderici) [inline spec/classes/init_spec.rb:225]: Thanks for adding those lost ones

**Comment by albgus** [comment]
The problem with distributing the certificate is that the private key would need to be distributed as well, and I'm not sure that it's a good idea to manage sensitive data in Puppet at the moment.

**Comment by Phil-Friderici** [comment]
@gusson truely, the functionality shouldn't be obligatory ;)
It would just feel more complete with an option to also distribute them.

**Comment by Phil-Friderici** [comment]
@gusson Thank you ! Released in v1.3.1

**Pull request merged**